man/demi_wait.md

```diff
@@ -33,10 +38,17 @@ the calling thread to block (spin) until the timeout `abstime` expires.
 specified by the list of queue tokens `qts` and it has a length of `num_qts`. This system call may cause the calling
 thread to block (spin) indefinitely.
 
+`demi_timedwait_any()` waits for the first asynchronous I/O operation in a set to complete or for the expiration of a
+timeout, whichever happens first. The set of I/O operations is specified by the list of queue tokens `qts` and it has a
+length of `num_qts`.  The `abstime` parameter specifies an absolute timeout in seconds and nanoseconds since the
+Epoch. If the I/O operation has already completed when `demi_timedwait()` is called, then this system call never fails
```

Comment:
Why is the timeout an absolute rather than a relative time.  The Unix API call our `demi_wait`* calls are most similar to (and will likely be replacing in a least some code we'd like to port over) is `select()`.  And `select` takes an _interval_ timeout, not an absolute one.

I'll also note that `struct timespec` evolved from `struct timeval`, which was initially introduced to the Unix programming world specifically to be the timeout interval for `select`.  So, I think most programmers will expect this to be a time interval, rather than an absolute time.

Reply:
The rationale for using absolute time was to follow other Linux APIs that alternatively to `select()` and `pselect()` rely

The rationale for using an absolute time in the first place:
- Other Linux APIs work with absolute time (ie: `epoll()` and `pthread_cond_timedwait()`) use absolute time.
- `timedwait()` was first implemented using absolute time and I did not aim to diverge from an existing pattern.

In my understanding the approach that you suggested and the one that I proposed work similarly, which of each having their pros and cons.

I happy to take this suggestion to the others and have a discussion. However, I will advocate for handling these semantic changes in a different PR.

Reply:
Um, `timedwait()` made it into the code base via a PR that skirted the proper review process, therefore I don't think we should consider it a precedent, but rather something that we should look at with great suspicion.  My preference would be to reverse that PR and have it go through the proper review process (so we could fix it and this one together).

As to what we should do here, I don't see the point of adding a new API call just to remove it again in a subsequent PR, especially when we can just fix it now by simply adding the timeout value to `demi_wait()` and `demi_wait_any()`.

Reply:
An additional argument for interval rather than absolute time is that it inherently more portable.  An absolute time requires an external definition (such as the one we use here, which relies upon the Unix Epoch).  An interval time does not.

Windows expresses absolute time quite differently from traditional Unix and has a different Epoch.  Now there is plenty of originally Unix code that has been ported to Windows that knows how to translate between the two representations, and there is probably even some Windows library support for this (I know at least .NET does).  So, Windows users can certainly make do with a Unix-centric API.  But why tie our API to Unix if we don't need to?

Reply:
The simplest API to use is often the best. If using `interval` is going to result in less code/responsibility on the developer (caller's part), then that sounds like a better solution to me. However, we can do it as a separate change (provided that is prioritized to happen soon) and agree upon these conventions for future APIs as well (if applicable). Also, we should document these in the guidelines that we all will agree on going forward.